src/org/labkey/test/tests/SampleSetTest.java

```diff
@@ -617,6 +633,85 @@ public void deleteLineageParent() throws IOException, CommandException
     }
 
+
+    @Test
+    public void testStringLookupFields() throws IOException, CommandException
+    {
+        String sampleSetName = "10000Samples"; // Testing with 10,000 samples because as per the product the lookup is converted into text field only when the samples exceed 10,000 samples
+        String listName = "MainList";
+
+        goToProjectHome();
+        new PortalHelper(this).addWebPart("Lists");
+
+        log("Creating the sample set of 10000 samples");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSetName);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("label", FieldDefinition.ColumnType.String)));
+        dgen.addDataSupplier("label", () -> dgen.randomString(10))
+                .withGeneratedRows(10000);
+        dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        SaveRowsResponse saveRowsResponse = dgen.insertRows(createDefaultConnection(true), dgen.getRows());
+        log("Successfully  inserted " + saveRowsResponse.getRowsAffected());
+
+        log("Waiting for the sample data to get generated");
+        goToProjectHome();
+        waitAndClickAndWait(Locator.linkWithText(sampleSetName));
+
+        log("Inserting 10,001 row in the sampleset");
+        DataRegionTable table = new DataRegionTable("Material", getDriver());
+        table.clickInsertNewRow();
+
+        setFormElement(Locator.name("quf_Name"), "Sample1");
+        setFormElement(Locator.name("quf_label"), "Sample1");
+        clickButton("Submit");
+
+        log("Creating the list");
+        goToProjectHome();
+        lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen1 = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String)));
+
+        DomainResponse createResponse = dgen1.createDomain(createDefaultConnection(true), "VarList", Map.of("keyName", "id"));
+
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        log("Adding the lookUp field in the list");
+        DomainFieldRow lookUpRow = domainFormPanel.addField("lookUpField")
+                .setType(FieldDefinition.ColumnType.Lookup)
+                .expand()
+                .setFromFolder("Current Folder")
+                .setFromSchema("exp.materials")
+                .setFromTargetTable("10000Samples (Integer)")
+                .setDescription("LookUp in same container with 10000 samples");
+
+        domainDesignerPage.clickFinish();
```

Comment:
Add this field during the initial list creation. No need to mess around in the UI for this test.